vaxrank/core_logic.py

```diff
@@ -49,6 +49,16 @@ def vaccine_peptides_for_variant(
         variant=variant,
         protein_sequence=isovar_protein_sequences[0])
 
+    min_expected_length = mhc_predictor.min_peptide_length
+    def_expected_length = mhc_predictor.default_peptide_lengths
+    if def_expected_length is not None and len(def_expected_length) > 0:
```

Comment:
ah, yeah those variable names - glad you bring that up, Julia because I have an interesting observation from late last night, where I had to spend close to an hour debugging/tracing this from vaxrank down to mhctools to make sure that what I understood about them was correct (since it is not obvious at first): 

 - `min` is  `1` for virtually all predictor implementations ([usage across hammerlab](https://github.com/search?utf8=%E2%9C%93&q=org%3Ahammerlab+min_peptide_length%3D&type=Code) and it actually can be dropped from the whole class since we never use it the way we should. Note that the project that has this search string is actually creating a different class. The funny part is almost all of the stack has implemented checks on this and also the max, but they are never informative.

- `max`: similarly but in a different way, it is always `None` and never properly used. I am assuming that this is just meant to be like this to represent unknown limits or no limit. I would like to take this opportunity to congraluate @timodonnell, since he is the only one that actually makes use of this variable in the `neon` and `mhcflurry` concep (and @iskandr earns the recent user badge since he is using it in `pepnet`): [max in the wild](https://github.com/search?utf8=%E2%9C%93&q=org%3Ahammerlab+max_peptide_length%3D&type=Code)  

- now comes the poorest variable of all time,`default, who gets pushed around quite a lot and has been discriminated due to its name: [see its usage across multiple hammerlab pygenomics projects](https://github.com/search?q=org%3Ahammerlab+default_peptide_lengths%3D&type=Code) From `mhctools` to `vaxrank` all of our pygenomics stacks uses it to store user preferences that are often parsed out of the command line arguments. 

Given that the `_check_peptide_lengths` method doesn't use it, smallish peptides were always falling through the cracks until `mhctools` naively tries to use it for prediction without knowing that is too small and returns an empty set up the chain (see topiary below):

```shell
$ topiary --mhc-predictor netmhccons --mhc-alleles HLA-A02:01 --genome GRCh37 --variant 3 53925802 G T --mhc-epitope-lengths 9
Topiary commandline arguments:
Namespace(genome='GRCh37', ic50_cutoff=None, json_variants=[], maf=[], mhc_alleles='HLA-A02:01', mhc_alleles_file=None, mhc_epitope_lengths=[9], mhc_peptide_lengths=None, mhc_predictor='netmhccons', only_novel_epitopes=False, output_csv=None, output_csv_sep=',', output_html=None, padding_around_mutation=None, percentile_cutoff=None, print_columns=False, rename_output_column=None, rna_gene_fpkm_tracking_file=None, rna_min_gene_expression=0.0, rna_min_transcript_expression=0.0, rna_transcript_fpkm_gtf_file=None, rna_transcript_fpkm_tracking_file=None, skip_variant_errors=False, subset_output_columns=None, variant=[['3', '53925802', 'G', 'T']], vcf=[], wildtype_ligandome_directory=None)
['netMHCcons']
WARNING:mhctools.base_predictor:Deprecated method 'predict', use 'predict_subsequences
Traceback (most recent call last):
  File "/home/arman/p/miniconda3/envs/bfx/bin/topiary", line 58, in <module>
    main(args)
  File "/home/arman/p/miniconda3/envs/bfx/bin/topiary", line 43, in main
    epitopes = predict_epitopes_from_args(args)
  File "/home/arman/p/miniconda3/envs/bfx/lib/python2.7/site-packages/topiary/predict_epitopes.py", line 294, in predict_epitopes_from_args
    raise_on_variant_effect_error=not args.skip_variant_errors)
  File "/home/arman/p/miniconda3/envs/bfx/lib/python2.7/site-packages/topiary/predict_epitopes.py", line 261, in predict_epitopes_from_variants
    wildtype_ligandome_dict=wildtype_ligandome_dict)
  File "/home/arman/p/miniconda3/envs/bfx/lib/python2.7/site-packages/topiary/predict_epitopes.py", line 157, in predict_epitopes_from_mutation_effects
    binding_predictions = mhc_model.predict(protein_subsequences)
  File "/home/arman/p/miniconda3/envs/bfx/lib/python2.7/site-packages/mhctools/base_predictor.py", line 147, in predict
    return self.predict_subsequences(sequence_dict, peptide_lengths=None)
  File "/home/arman/p/miniconda3/envs/bfx/lib/python2.7/site-packages/mhctools/base_predictor.py", line 128, in predict_subsequences
    binding_predictions = self.predict_peptides(sorted(peptide_set))
  File "/home/arman/p/miniconda3/envs/bfx/lib/python2.7/site-packages/mhctools/base_commandline_predictor.py", line 309, in predict_peptides
    temp_dir_list=dirs)
  File "/home/arman/p/miniconda3/envs/bfx/lib/python2.7/site-packages/mhctools/base_commandline_predictor.py", line 256, in _run_commands_and_collect_predictions
    process_limit=self.process_limit)
  File "/home/arman/p/miniconda3/envs/bfx/lib/python2.7/site-packages/mhctools/process_helpers.py", line 115, in run_multiple_commands_redirect_stdout
    assert len(multiple_args_dict) > 0
AssertionError
```

If I happen to find some time, I can refactor the stack and make better use of these arguments, but that is a long term shot for me now ;)